agent/agent.go

```diff
@@ -590,16 +578,10 @@ func (a *Agent) initializeConnectionPool() error {
 		rpcSrcAddr = &net.TCPAddr{IP: a.config.RPCBindAddr.IP}
 	}
 
-	// Ensure we have a log output for the connection pool.
-	logOutput := a.LogOutput
-	if logOutput == nil {
-		logOutput = os.Stderr
-	}
-
 	pool := &pool.ConnPool{
 		Server:          a.config.ServerMode,
 		SrcAddr:         rpcSrcAddr,
-		LogOutput:       logOutput,
+		Logger:          a.logger.StandardLogger(&hclog.StandardLoggerOptions{InferLevels: true}),
```

Comment:
I've used this setup of a stdlib `log.Logger` in a few places. I'm not sure if we need to name these new loggers. I guess previously they would not have had a name. If anyone knows how we should name these loggers being passed to yamux I can add a `Named()`.